server/channels/app/imaging/utils.go

```diff
@@ -140,8 +142,156 @@ func FillImageTransparency(img image.Image, c color.Color) {
+// Adapted from github.com/disintegration/imaging
+func CropCenter(img image.Image, w, h int) image.Image {
+	srcBounds := img.Bounds()
+	anchorPoint := image.Pt(srcBounds.Min.X+(srcBounds.Dx()-w)/2, srcBounds.Min.Y+(srcBounds.Dy()-h)/2)
+	r := image.Rect(0, 0, w, h).Add(anchorPoint)
+	b := srcBounds.Intersect(r)
+	return transform.Crop(img, b)
+}
+
+// cropAndResize crops the image to the smallest possible size that has the required aspect ratio using
+// a centered anchor point, then scales it to the specified dimensions and returns the transformed image.
+//
+// This is generally faster than resizing first, but may result in inaccuracies when used on small source images.
+// Adapted from github.com/disintegration/imaging
+func cropAndResizeCenter(img image.Image, width, height int) image.Image {
+	dstW, dstH := width, height
+
+	srcBounds := img.Bounds()
+	srcW := srcBounds.Dx()
+	srcH := srcBounds.Dy()
+	srcAspectRatio := float64(srcW) / float64(srcH)
+	dstAspectRatio := float64(dstW) / float64(dstH)
+
+	var tmp image.Image
+	if srcAspectRatio < dstAspectRatio {
+		cropH := float64(srcW) * float64(dstH) / float64(dstW)
+		tmp = CropCenter(img, srcW, int(math.Max(1, cropH)+0.5))
+	} else {
+		cropW := float64(srcH) * float64(dstW) / float64(dstH)
+		tmp = CropCenter(img, int(math.Max(1, cropW)+0.5), srcH)
+	}
+
+	return Resize(tmp, dstW, dstH, transform.Lanczos)
+}
+
+// resizeAndCrop resizes the image to the smallest possible size that will cover the specified dimensions,
+// crops the resized image to the specified dimensions using a centered anchor point and returns
+// the transformed image.
+// Adapted from github.com/disintegration/imaging
+func resizeAndCropCenter(img image.Image, width, height int) image.Image {
+	dstW, dstH := width, height
+
+	srcBounds := img.Bounds()
+	srcW := srcBounds.Dx()
+	srcH := srcBounds.Dy()
+	srcAspectRatio := float64(srcW) / float64(srcH)
+	dstAspectRatio := float64(dstW) / float64(dstH)
+
+	var tmp image.Image
+	if srcAspectRatio < dstAspectRatio {
+		tmp = Resize(img, dstW, 0, transform.Lanczos)
+	} else {
+		tmp = Resize(img, 0, dstH, transform.Lanczos)
+	}
+
+	return CropCenter(tmp, dstW, dstH)
+}
+
 // FillCenter creates an image with the specified dimensions and fills it with
 // the centered and scaled source image.
-func FillCenter(img image.Image, w, h int) *image.NRGBA {
-	return imaging.Fill(img, w, h, imaging.Center, imaging.Lanczos)
+// To achieve the correct aspect ratio without stretching, the source image will be cropped.
+// Adapted from github.com/disintegration/imaging
+func FillCenter(img image.Image, dstW, dstH int) image.Image {
+	if dstW <= 0 || dstH <= 0 {
+		return &image.RGBA{}
+	}
+
+	srcBounds := img.Bounds()
+	srcW := srcBounds.Dx()
+	srcH := srcBounds.Dy()
+
+	if srcW <= 0 || srcH <= 0 {
+		return &image.RGBA{}
+	}
+
+	if srcW == dstW && srcH == dstH {
+		return clone.AsShallowRGBA(img)
+	}
+
+	if srcW >= 100 && srcH >= 100 {
```

Comment:
Where are these numbers coming from? Can we refactor to named constants to clearly state their purpose?